nucypher/policy/maps.py

```diff
@@ -103,329 +205,112 @@ def from_bytes(cls, data: bytes):
 
 
-class TreasureMapSplitter(BrandingMixin, VersioningMixin, BytestringKwargifier):
-    pass
-
+class EncryptedTreasureMap:
 
-class TreasureMap:
-    VERSION_NUMBER = 1  # Increment when serialization format changes.
-
-    _BRAND = b'TM'
-    _VERSION = int(VERSION_NUMBER).to_bytes(VersioningMixin.HEADER_LENGTH, 'big')
-
-    class NowhereToBeFound(RestMiddleware.NotFound):
-        """
-        Called when no known nodes have it.
-        """
-
-    class IsDisorienting(Exception):
-        """
-        Called when an oriented TreasureMap lists fewer than m destinations, which
-        leaves Bob disoriented.
-        """
-
-    class OldVersion(Exception):
-        """Raised when a treasure map's version is too old or contents are incompatible."""
-
-    ursula_and_kfrag_payload_splitter = BytestringSplitter(
-        (to_checksum_address, ETH_ADDRESS_BYTE_LENGTH),
+    _splitter = BytestringSplitter(
+        signature_splitter,
+        hrac_splitter,
         (UmbralMessageKit, VariableLengthBytestring),
-    )
+        (bytes, 1))
 
     from nucypher.crypto.signing import \
         InvalidSignature  # Raised when the public signature (typically intended for Ursula) is not valid.
 
-    def __init__(self,
-                 m: int = None,
-                 destinations=None,
-                 message_kit: UmbralMessageKit = None,
-                 public_signature: Signature = None,
-                 hrac: Optional[bytes] = None,
-                 version: bytes = None
-                 ) -> None:
-
-        if version is not None:
-            self.version = version
-
-        if m is not None:
-            if m > 255:
-                raise ValueError("Largest allowed value for m is 255.")
-            self._m = m
-
-            self._destinations = destinations or {}
-        else:
-            self._m = NO_DECRYPTION_PERFORMED
-            self._destinations = NO_DECRYPTION_PERFORMED
-
-        self._id = None
-
-        self.message_kit = message_kit
-        self._public_signature = public_signature
-        self._hrac = hrac
-        self._payload = None
-
-        if message_kit is not None:
-            self.message_kit = message_kit
-            self._set_id()
-        else:
-            self.message_kit = None
-
-    def __eq__(self, other):
-        try:
-            return self.public_id() == other.public_id()
-        except AttributeError:
-            raise TypeError(
-                f"Can't compare {type(other).__name__} to a TreasureMap (it needs to implement public_id() )")
-
-    def __iter__(self):
-        return iter(self.destinations.items())
-
-    def __len__(self):
-        return len(self.destinations)
-
-    def __bytes__(self):
-        if self._payload is None:
-            self._set_payload()
-        return self._BRAND + self._VERSION + self._payload
+    @staticmethod
+    def _make_blockchain_signature(blockchain_signer: Callable[[bytes], bytes],
```

Comment:
```suggestion
    def _sign(blockchain_signer: Callable[[bytes], bytes],
```

Perhaps this can be combined with the federated signing method too, appending all zeros to match the format.